src/messages/mod.rs

```diff
@@ -559,144 +554,97 @@ mod tests {
-        let full_id_3 = FullId::gen(&mut rng);
+        let sk_share_0 = SectionKeyShare::new_with_position(0, sk_set.secret_key_share(0));
+        let sk_share_1 = SectionKeyShare::new_with_position(1, sk_set.secret_key_share(1));
 
-        let bls_keys = generate_bls_threshold_secret_key(&mut rng, 4);
-        let bls_secret_key_share_0 =
-            SectionKeyShare::new_with_position(0, bls_keys.secret_key_share(0));
-        let bls_secret_key_share_3 = bls_keys.secret_key_share(3);
+        let msg = gen_message(&mut rng);
+        let proof = make_proof_chain(&pk_set);
 
-        let socket_addr: SocketAddr = unwrap!("127.0.0.1:9999".parse());
-        let connection_info = ConnectionInfo {
-            peer_addr: socket_addr,
-            peer_cert_der: vec![],
-        };
-        let content = (0..10).collect();
-        let name: XorName = rand::random();
-        let msg = RoutingMessage {
-            src: Authority::Section(name),
-            dst: Authority::Section(name),
-            content: MessageContent::UserMessage(content),
-        };
-
-        let prefix = Prefix::new(0, *full_id_0.public_id().name());
-        let src_section_nodes = vec![
-            P2pNode::new(*full_id_0.public_id(), connection_info.clone()),
-            P2pNode::new(*full_id_1.public_id(), connection_info.clone()),
-            P2pNode::new(*full_id_2.public_id(), connection_info.clone()),
-            P2pNode::new(*full_id_3.public_id(), connection_info),
-        ];
-        let src_section = unwrap!(EldersInfo::new(
-            src_section_nodes
-                .into_iter()
-                .map(|node| (*node.public_id().name(), node))
-                .collect(),
-            prefix,
-            None,
+        // Message with valid signature
+        let mut signed_msg_0 = unwrap!(SignedRoutingMessage::new(
+            msg.clone(),
+            &sk_share_0,
+            pk_set.clone(),
+            proof.clone()
         ));
-        let pk_set = bls_keys.public_keys();
-        let dummy_key_info = SectionKeyInfo::from_elders_info(&src_section, pk_set.public_key());
-        let dummy_proof = SectionProofChain::from_genesis(dummy_key_info);
-        let mut signed_msg = unwrap!(SignedRoutingMessage::new(
-            msg,
-            &bls_secret_key_share_0,
+
+        // Message with invalid signature
+        let invalid_signature_share = sk_share_1.key.sign(b"bad message");
+        let metadata = SecurityMetadata::Partial(PartialSecurityMetadata {
             pk_set,
-            dummy_proof
-        ));
-        assert_eq!(signed_msg.signatures().expect("no signatures").len(), 1);
-        assert!(!signed_msg.check_fully_signed());
-
-        // Add an invalid signature for IDs 1 added by the 3rd malicious node.
-        // Add a valid signature for IDs 1 and 2 and an invalid one for ID 3
-        // Add an invalid signature for ID 3 added by the same 3rd malicious node.
-        let bad_sig = bls_secret_key_share_3.sign(&[1]);
```

Comment:
Is this part of the test no longer relevant (i.e testing that adding bad signatures do not make the signature invalid if enough good ones are present?

Reply:
There is a test that invalid signatures don't count, but not that they don't invalidate valid signatures. I'll add it.